server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -2575,6 +2576,44 @@ protected void runInContext() {
         }
     }
 
+    private void verifyVmLimits(UserVmVO vmInstance, Map<String,String> details) {
```

Comment:
@Pearl1594 
do we need to check cpu/memory if vm uses constraint service offering ?

Reply:
I think you are right @weizhouapache , we'll need to add a check to verify if the modified values for cpuNumber / memory are within the limits of the custom offering. Thanks - I'll add the check